File: app/src/main/java/me/waister/qualcompensa/HowWorksActivity.java
package me.waister.qualcompensa;

import android.app.Activity;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatButton;
import android.view.View;

public class HowWorksActivity extends AppCompatActivity {

    private Activity mActivity = HowWorksActivity.this;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_how_works);

        AppCompatButton buttonSubmit = (AppCompatButton) findViewById(R.id.button_submit);

        if (buttonSubmit != null) {
            buttonSubmit.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    SharedPreferences mPreferences = Pref.getPreferences(mActivity);
                    SharedPreferences.Editor editor = mPreferences.edit();
                    editor.putBoolean(Pref.FIRST_ACCESS, false);
                    editor.apply();

                    onBackPressed();
                }
            });
        }
    }
}

File: app/src/main/java/me/waister/qualcompensa/activity/HowWorksActivity.kt
package me.waister.qualcompensa.activity

import android.graphics.Color
import android.os.Bundle
import androidx.activity.SystemBarStyle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updatePadding
import com.orhanobut.hawk.Hawk
import me.waister.qualcompensa.databinding.ActivityHowWorksBinding
import me.waister.qualcompensa.utils.PREF_FIRST_ACCESS

class HowWorksActivity : AppCompatActivity() {

    private lateinit var binding: ActivityHowWorksBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge(
            statusBarStyle = SystemBarStyle.dark(Color.TRANSPARENT),
            navigationBarStyle = SystemBarStyle.dark(Color.TRANSPARENT)
        )
        super.onCreate(savedInstanceState)

        binding = ActivityHowWorksBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.buttonSubmit.setOnClickListener {
            Hawk.put(PREF_FIRST_ACCESS, false)

            onBackPressedDispatcher.onBackPressed()
        }

        setupCommonInsets()
    }

    private fun setupCommonInsets() = with(binding) {
        ViewCompat.setOnApplyWindowInsetsListener(layoutMain) { view, insets ->
            val bars =
                insets.getInsets(WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout())
            view.apply {
                updatePadding(left = bars.left, top = bars.top, right = bars.right, bottom = bars.bottom)
            }
            insets
        }
    }

}
